File: event_creator/course.c
/*
 * course.c
 *
 * Tom Leaman (thl5@aber.ac.uk)
 */

#include <stdio.h>
#include <stdlib.h>

#include "data.h"
#include "dbg.h"
#include "util.h"

Course* Course_new(const char id) {
  Course* course = malloc(sizeof(Course));
  check_mem(course);

  course->id = id;
  course->num_nodes = 0;
  course->nodes = Vector_new(sizeof(int), NULL);

  return course;
error:
  exit(EXIT_FAILURE);
}

void Course_destroy(Course* course) {
  if (course) {
    if (course->nodes) Vector_destroy(course->nodes);
    free(course);
  }
}

void Course_add_node(Course* course, int node_id) {
  Vector_add(course->nodes, &node_id);
}

int Course_write(Course* course, const char* filename) {
  FILE* fp;
  int node;
  int i;

  fp = fopen(filename, "a");
  check(fp != NULL, "Could not open %s for writing", filename);
  fprintf(fp, "%c %i", course->id, course->num_nodes);
  for (i = 0; i < course->num_nodes; i++) {
    Vector_get(course->nodes, i, &node);
    fprintf(fp, " %i", node);
  }
  fprintf(fp, "\n");
  fclose(fp);

  return 0;
error:
  exit(EXIT_FAILURE);
}

File: checkpoint_manager/src/event/Course.java
package event;

import event.node.Node;

import java.util.List;

/**
 * 
 * @author Tom Leaman (thl5@aber.ac.uk)
 *
 */
public class Course {

	private char id;
	private List<Node> nodes;

	public Course(char id, List<Node> nodes) {
		this.id = id;
		this.nodes = nodes;
	}

	public char getId() {
		return id;
	}

	public List<Node> getNodes() {
		return nodes;
	}

	public Node getLastNode() {
		return nodes.get(nodes.size() - 1);
	}

}
